### DISCUSSION TITLE
Still having trouble getting the synchronizer setup

### DISCUSSION THREAD

**Discussion by MathyFey** [GameNetworking/network_synchronizer#115] [General]
Hi, so from what I figured out so far about the problem I was having at first, is that.
The player character needs to be registered before adding the character to the scene. by doing this we prevent the crash, but another weird thing happens.

Server error log:

![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/2282f8f2-1fe3-42c3-85ae-acb9f50189cc)

Client error log:

![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/5116ffc4-73af-4003-a6a0-447926ea77ee)

The errors just keep repeating

So describing what I'm doing would be kind of like this:
Here's a few login so you don't need to make one (user: admin, pass: admin) and (user:test, pass: test)
1. Database stuff, you can just ignore its not tied to the logic at all
2. When we click on the server button we create the server and start waiting for clients to connect
3. When we click on the client button we just connect to the server, nothing else
4. When we click on the Start Match button though:
    1. Instantiate a character for each 
    2. Set the character's name to their id
    3. Register the character
    4. Add to the tree
    5. Set the global position to one of the spawn points
5. On the OnReady of the character I set their authority to their name
    
   All that is working as you can see here the characters have the proper names and authority:
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/c0989c02-45ce-4b28-840e-cd78151dac80)
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/674695a2-9a03-4cc1-ac83-de3d60479438)

Server View: (Server character can move just fine)
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/6e12636c-d9be-48f3-bf71-18413a5ca17d)

Client View: (All the characters even the server's are frozen)
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/bb3862a2-521b-456e-867e-0a15c944297a)

The server can move just fine and but the client is frozen in time kind of
I don't think the controller process is being called at all on anything except the server character

I made another commit to the repo I posted earlier with this project, tho unlike the others this is kind of part of the game I'm making and not just testing now, so its a bit more convoluted, sorry about that e.e .

https://github.com/MathyFey/NetworkSyncTest

But please help, your module is pretty much the last hope I have for my game, I don't want to get kicked back to unreal...

**Comment by MathyFey** [comment]
Nevermind about registering before, I flipped it again and it still didn't crash, I have no idea why that crash happened anymore...

New logs from this last try:

Server:
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/fc4f1645-c981-463c-abe4-a5c41c7b018a)

Client:
![image](https://github.com/GameNetworking/network_synchronizer/assets/45970379/df6858da-15a7-4143-b63e-562e85a9aa53)

Sorry If I'm not making much sense, I tried to get the point across but I'm sick rn, fever and stuff like that e.e

**Reply by Malkverbena** [reply]
Did you fix this? I have the same issue.

**Reply by AndreaCatania** [reply]
Can you please show the code you are using for registering it and the crash log?

**Comment by Malkverbena** [comment]
This is the code I'm trying to use.
The system is Ubuntu 22.04
[NetTest.zip](https://github.com/GameNetworking/network_synchronizer/files/15421089/NetTest.zip)

**Reply by Malkverbena** [reply]
I just made another try with a different code but I'm still getting the same.
The error occurs on clients. Apparently can't find the themselves`s player.
Should I keep it in a separated  issue?

**Comment by Malkverbena** [comment]


**Reply by AndreaCatania** [reply]
Hey Malkverbena, I'm out and I can't try your code right now. I'll be back next week and I'll be able to help you.

When you instantiate the SceneSync don't forget to call the function setup, that will ensure to properly begin the singleton

**Reply by Malkverbena** [reply]
The function setup you say is ¨_setup_synchronizer" ?

**Reply by AndreaCatania** [reply]
Try doing this and if doesn't work, send me the updated sample and I'll make sure to fix the issue.
On the server side, once the server is created call `reset_synchronizer_mode();`.
On the client side, once the client is connected to the server call `reset_synchronizer_mode();`.

Let me know if that fixes the issue.

**Comment by Malkverbena** [comment]
Hi, I beleve I followed all the steps.
But I could not make it work.


[NetTest.zip](https://github.com/user-attachments/files/15526361/NetTest.zip)
